src/app/article/[id]/page.jsx

```diff
@@ -23,10 +23,20 @@ export async function generateMetadata({ params }) {
 }
 
 export async function generateStaticParams() {
-  const items = await getAllContentList();
-  const paramID = items.map((item) => ({
-    id: item.topics_id.toString(),
-  }));
+  let page = 1;
+  const paramID = [];
+  let pageLimit;
+
+  do {
+    const { list, pageInfo } = await getAllContentList(page);
+    pageLimit = pageInfo.totalPageCnt;
+    const ids = list.map((item) => ({
+      id: item.topics_id.toString(),
+    }));
+    paramID.push(...ids);
+    page++;
+  } while (page <= pageLimit);
```

Comment:
I think this code looks good though,
Could you refactor this to be a recursive function?

There are 2 points I think
- by my personal experience I feel `do...while` statement is a bit dangerous because of relying mutable data
- recursive fn has an advantage that does not need to have temporary 
vars: `page, paramID, pageLimit`.

BTW about `for` loop statement vs recursive loop fn,
both have pros and cons,
so we can choose appropriate way.

As opposite `do...while` easily becomes a pitfall such as infinite loop bug which is fatal (would break browser).

This code perfectly fine but since this repository would be a reference for other devs,
I think we should avoid using `while...do` statement.

Reply:
my suggestion is like below (not tested)
```js
export default async function getAllContentList(page = 1) {
  const url = new URL(
    `${process.env.NEXT_PUBLIC_BASE_URL}/rcms-api/1/content/all`,
  );
  url.searchParams.append('pageID', page);

  const { list, pageInfo } = await fetch(url);

  const hasMorePage = pageInfo.page < pageInfo.totalPageCnt;
  return [
    ...list,
    ...(hasMorePage ? await getAllContentList(page + 1) : [])
  ]
}
```

Reply:
@yabe-diverta san 

Thank you for your insights. 🙇 

I agree with all your points. Let us switch to **recursion** and change the point of focus from `generateStaticParams` to `getAllContentList` function call. 

I have a suggestion to your recursion function. Instead of returning list, we could only return the topicsID instead as this would drastically reduce the memory space required for large dataset size.

```js
export default async function getAllContentList(page = 1) {
  const url = new URL(
    `${process.env.NEXT_PUBLIC_BASE_URL}/rcms-api/1/content/all`,
  );
  url.searchParams.append('pageID', page);

  const { list, pageInfo } = await fetch(url).then(response => response.json());
  const hasMorePage = pageInfo.pageNo < pageInfo.totalPageCnt;
  const currentTopicsIds = list.map((item) => ({
    id: item.topics_id.toString(),
  }));

  return [
    ...currentTopicsIds,
    ...(hasMorePage ? await getAllContentList(page + 1) : []),
  ];
} 
```

I again confirmed its working here using perPage = 10 https://diverta.gyazo.com/ed27a0e75344157df7be7498821cc82c

Your thoughts ?

Reply:
@VortexExpansion 
Yes looks better (unless other components don't use that function).
In that case would you rename the function? like `getAllContentIDs`.

Reply:
@yabe-diverta san 
Sorry, I checked that sitemap uses this function. 
Should we let this be as it is then or do some other workaround ?

Reply:
@VortexExpansion 
Hmm I think `let this be` could be OK because other new components will want to use that function as a general API call.
Even though I still don't have a strong opinion so either is fine.

Reply:
@yabe-diverta san 
Fair enough. Preserving the behaviour of `getAllContentList()` for now.